customer-commons/src/test/java/no/unit/nva/customer/model/CustomerDtoTest.java

```diff
@@ -24,11 +31,49 @@ class CustomerDtoTest {
     @Test
     void dtoSerializesToJsonAndBack() throws BadRequestException {
         CustomerDto customer = randomCustomer();
+        customer.getDoiAgent().addSecret("****");
         assertThat(customer, doesNotHaveEmptyValues());
         var json = customer.toString();
         var deserialized = CustomerDto.fromJson(json);
         assertThat(deserialized, is(equalTo(customer)));
         assertThat(deserialized, doesNotHaveEmptyValues());
+        assertEquals(deserialized.hashCode(), customer.hashCode());
+        assertNotEquals(null, deserialized);
+
+        json = customer.getDoiAgent().toString();
+        var deserializedDoiAgent = DoiAgentDto.fromJson(json);
+
+        assertEquals(deserializedDoiAgent.hashCode(), customer.getDoiAgent().hashCode());
+        assertNotEquals(null, deserializedDoiAgent);
+    }
+
+    @Test
+    void validJsonFromLinkItem() throws IOException {
+        var linkItem = attempt(() -> new LinkItem(randomUri().toURL().toString())).get();
+        var item = JsonConfig.readValue(linkItem.toString(), LinkItem.class);
+        assertThat(item.toString(), is(equalTo(linkItem.toString())));
+    }
+
+    @Test
+    void shouldThrowBadRequestWhenFailingToDeserializeDoiAgent() {
+        String invalidJson = randomString();
+        Executable action = () -> DoiAgentDto.fromJson(invalidJson);
+        var exception = assertThrows(BadRequestException.class, action);
+        assertThat(exception.getMessage(), containsString(invalidJson));
+    }
+
+    @Test
+    void dtoSerializesToJsonAndBackWithSecret() throws BadRequestException {
+        CustomerDto customer = randomCustomer();
+        customer.getDoiAgent().addSecret("******");
+        var json = customer.toString();
+        var deserialized = CustomerDto.fromJson(json);
+
+        var doi = deserialized.getDoiAgent();
+        assertThat(doi.toString(), doesNotHaveEmptyValues());
+        assertEquals(doi, deserialized.getDoiAgent());
```

Comment:
Are you comparing two equal objects? See `var doi = deserialized.getDoiAgent();` Are you meaning to compare with `customer.getDoiAgent()` ?